#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_matplotlibrc_path_search_fix.dpatch by Sandro Tosi <morph@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes the path to search for matplotlibrc file

@DPATCH@

Index: matplotlib-0.98.3/lib/matplotlib/__init__.py
===================================================================
--- matplotlib-0.98.3.orig/lib/matplotlib/__init__.py	2008-08-22 11:52:37.253904720 +0200
+++ matplotlib-0.98.3/lib/matplotlib/__init__.py	2008-08-22 11:52:42.250189441 +0200
@@ -452,9 +452,11 @@
             raise RuntimeError('Path in environment MATPLOTLIBDATA not a directory')
         return path
 
-    path = os.sep.join([os.path.dirname(__file__), 'mpl-data'])
+    path = '/usr/share/matplotlib/mpl-data'
     if os.path.isdir(path): return path
-
+    
+    raise RuntimeError('Could not find the matplotlib data files')
+    
     # setuptools' namespace_packages may highjack this init file
     # so need to try something known to be in matplotlib, not basemap
     import matplotlib.afm
@@ -549,7 +551,7 @@
     if os.path.exists(fname): return fname
 
 
-    path =  get_data_path() # guaranteed to exist or raise
+    path =  '/etc' # guaranteed to exist or raise
     fname = os.path.join(path, 'matplotlibrc')
     if not os.path.exists(fname):
         warnings.warn('Could not find matplotlibrc; using defaults')
